src/simplify_algebra.cpp

```diff
@@ -266,6 +266,84 @@ struct find_mul_dot
     }
 };
 
+struct find_dot_slice
```

Comment:
Minor: It might be more descriptive to name this matcher `find_dot_slice_used_once`.

Reply:
"Dot" should have one use only. 
Not the slice. it is not clear from the name `find_dot_slice_used_once`. We don't have "used_once"  inside names of any matchers